src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1267,6 +1274,20 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       nextUnitOfWork = null;
     }
 
+    if (
+      didCommit &&
+      (priorityLevel === TaskPriority || priorityLevel === SynchronousPriority)
+    ) {
+      invariant(
+        nestedSyncUpdates++ <= NESTED_SYNC_UPDATE_LIMIT,
+        '%s(...): Maximum update depth exceeded. This can happen when a ' +
+          'component repeatedly calls setState inside componentWillUpdate or ' +
+          'componentDidUpdate. React limits the number of nested updates to ' +
+          'prevent infinite loops.',
+        getComponentName(fiber),
```

Comment:
AFAIK our invariant transform does this automatically. But not warning. That said I'm all for explicitness.